all/test.sh

```diff
@@ -375,9 +381,29 @@ for lang in ${LANGS[@]}; do
     fi
 
     # Test without an output directory.
-    testGeneration "$lang" "$lang" "$expected_output_dir" 0
+    testGeneration "$lang" "$lang" "false" "$expected_output_dir" 0
 
     # Test with an output directory.
     test_dir="gen/foo/bar"
-    testGeneration "${lang}_with_output_dir" "$lang" "$test_dir" 0 -o "$test_dir"
+    testGeneration "${lang}_with_output_dir" "$lang" "false" "$test_dir" 0 -o "$test_dir"
 done
+
+#Generate proto files but using the -d switch to just pass a directory input
+for lang in ${LANGS[@]}; do
+    expected_output_dir=""
+    if [[ "$lang" == "python" ]]; then
+        expected_output_dir="gen/pb_$lang"
+    else
+      expected_output_dir="gen/pb-$lang"
+    fi
+
+    # Test without an output directory.
+    testGeneration "$lang" "$lang" "false" "$expected_output_dir" 0
+
+    # Test with an output directory.
+    test_dir="gen/foo/bar"
+    testGeneration "${lang}_from_dir_with_output_dir" "$lang" "true" "$test_dir" 0 -o "$test_dir"
+done
+
+#Test the -d for go but also add the the "--go-full-directory-depth" which removes the shallow find flag
+testGeneration "go_full_depth_search" "go" "true" "gen/pb-go" 0 "--go-full-directory-depth"
```

Comment:
- I think that for this test we should have multiple proto files in different directory depths and verify that they are all generated, and also that those files are NOT generated when the flag is not used.
 - If you don't mind also moving this test to where the other `go` specific tests are?